packages/core/bootstrap/src/lib/server.ts

```diff
@@ -124,3 +127,33 @@ function setupMetricsServer() {
 
   metricsApp.listen(metricsPort, () => logger.info(`Monitoring listening on port ${metricsPort}!`))
 }
+
+const windowMs = 1000 * 5
+const max = parseInt(process.env.SERVER_RATE_LIMIT_MAX || '250') // default to 250 req / 5 seconds max
+const delayAfter = max * (Number(process.env.SERVER_SLOW_DOWN_AFTER_FACTOR) || 0.8) // we start slowing down requests when we reach 80% of our max limit for the current interval
+const delayMs = parseInt(process.env.SERVER_SLOW_DOWN_DELAY_MS || '500') // default to slowing down each request by 500ms
+
+function initExpressMiddleware(app: express.Express) {
+  app.set('trust proxy', 1)
```

Comment:
Did we not set this previously? What's the reason we're setting it now?

Should this be configurable by the operator, so they can disable this if they're not actually behind a proxy?

Reply:
Did not set it previously. I added it because it is shown in the example for these middleware packages.

Based on some quick researching I didn't see a downside to leaving it on for everyone.
https://stackoverflow.com/questions/23413401/what-does-trust-proxy-actually-do-in-express-js-and-do-i-need-to-use-it

We could make it configurable if there is a risk, but otherwise the less environment variables to set the better.

Reply:
I see. Under the assumption that node ops have their networking set correctly, I guess it shouldn't be an issue.

If the adapters are reachable from the public however, this would allow attackers to spoof the requester IP address and essentially limit the legitimate requesters. But I think node ops are in bigger trouble if the adapters were public anyways.

Reply:
This scenario of spoofing IPs to get them rate limited won't actually even be relevant because of:
```
    keyGenerator: () => '*', // use one key for all incoming requests
    ```